lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -144,11 +145,15 @@ def set_functions(self, functions):
             If a transform is supplied, then the transform must have an
             inverse.
         """
+
         if (isinstance(functions, tuple) and len(functions) == 2 and
                 callable(functions[0]) and callable(functions[1])):
             # make an arbitrary convert from a two-tuple of functions
             # forward and inverse.
             self._functions = functions
+        elif isinstance(functions, Transform):
+            self._functions = (
+                    functions.transform, functions.inverted().transform)
```

Comment:
This is dangerous, as `Transform.inverted` returns a _copy_ with a snapshot of the transform, and thus won't be in sync if the input transform changes. Instead, you'd probably want (identity - input).

Reply:
I agree, but is it preferable to implicitly allow arguments to be mutated after the fact as a side effect at all? I.e., would a better way to ensure consistency be to simply copy the passed transform to begin with? Let me know which way to go (not sure if there's precedent here)

Reply:
Yes, we tend to be lazily evaluated, e.g., if we had `Axes.transData` here, it might change depending on limits (though of course that specific transform won't work here, being 2D, and as you've noted above this argument is 1D.)

Reply:
Actually, I'm not finding that to be the case (that subtraction avoids inconsistency upon mutation), at least with the `Translation` class I defined:

```python
import matplotlib.pyplot as plt
import matplotlib.transforms as mtransforms


class Translation(mtransforms.Transform):
    input_dims = 1
    output_dims = 1

    def __init__(self, dx):
        self.dx = dx

    def transform(self, values):
        return values + self.dx

    def inverted(self):
        return Translation(-self.dx)


forward = Translation(1)
backward = mtransforms.IdentityTransform() - forward
print(forward.transform(1), backward.transform(1))
forward.dx = 2
print(forward.transform(1), backward.transform(1))
```
prints
```
2 0
3 0
```
Reviewing [`Transform.__sub__`](https://github.com/matplotlib/matplotlib/blob/v3.8.1/lib/matplotlib/transforms.py#L1418-L1466), I can't actually say I understand how lazy inversion ever occurs... let me know what you think.

Reply:
But your test is wrong? The second printout should be `3 -1`, if `backward` and `forward` are linked (which is what we want here, since the user only supplied the forward transform.)

Reply:
That's my point - `forward` and `IdentityTransform() - forward` are not linked, and I don't see any mechanism for that operation being lazy in `Transform.__sub__`.

Reply:
Oh, I see what you're saying now. Yes, I misremembered what happened there. What you will have to do is put a `lambda` in that calculates the inverse when it's called.

Reply:
Clever - [2803407](https://github.com/matplotlib/matplotlib/pull/27267/commits/280340764ce97aa211b78a075def2a0b7cb9f33e)